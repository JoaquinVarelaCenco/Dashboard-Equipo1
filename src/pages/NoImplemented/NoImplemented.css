.noImplemented{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    gap: 15px;
    min-height: calc(100vh - 72px);
    margin-top: 72px;
    height: 100vh;
    width: 100%;
}

.noImplemented h3{
    font-size: 4.5rem;
    text-transform: uppercase;
}

.noImplemented h3 span{
    display: block;
}

.noImplemented p{
    font-size: 28px;
    font-weight: 300;
    text-transform: uppercase;
}

.progress{
    height: 20px;
    width: 30%;
    border: 1px solid #428bca;
    border-radius: 5px;
    background-color: #e6f3fa;
    margin-bottom: 15px;
}

.progress-bar{
    height: 100%;
    width: 75%;
    background: #428bca;
    display: flex;
    align-items: center;
    transition: width 0.25s;
    border-radius: 5px;
    animation-name: loopProgress;
    animation-duration: 6s;
    animation-iteration-count: infinite;
    animation-timing-function: ease-in-out;
}

@keyframes loopProgress {
    0%{
        width: 10%;
    }
    70%{
        opacity: 1;
    }

    100%{
        width: 90%;
        opacity: .2;
    }
}

.noImplemented button{
    background-color: #5b5d5d ;
    border: none;
    padding: 8px 10px;
    font-size: 16px;
    color: white;
    margin: 0 6px;
    border-radius: 15px;
    min-width: 150px;
}

@media ( max-width:560px){
    .noImplemented h3{
        font-size: 3rem;
    }

    .noImplemented p{
        font-size: 20px;
    }
}

@media ( max-width:360px){
    .noImplemented h3{
        font-size: 2rem;
    }

}