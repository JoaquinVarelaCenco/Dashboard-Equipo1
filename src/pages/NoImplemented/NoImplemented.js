import { Link } from 'react-router-dom'
import Button from '../../components/Button/Button'
import './NoImplemented.css'

const NoImplemented = () => {

  return (
    <div className={`noImplemented`}>
        <h3><span>En</span> Construcción</h3>
        <p>Esta sección estara lista pronto</p>
        <div className='progress'>
            <div className='progress-bar'>
            </div>
        </div>
        <Link to="/home">
            <Button 
              title={"Inicio"}
            />
        </Link>
    </div>
  )
}

export default NoImplemented