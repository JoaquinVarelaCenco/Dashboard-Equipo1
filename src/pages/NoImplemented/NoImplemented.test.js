import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NoImplemented from "./NoImplemented";

let component;
let btnInicio;

describe('Test de la vista NoImplemented', ()=>{

    beforeEach(()=>{
        component = render(<NoImplemented />, { wrapper: BrowserRouter })
        btnInicio = screen.getByRole('link');
    })

    it('La vista debe renderizarse correctamente', ()=>{
        const { container } = component;

        expect(container).toMatchSnapshot();
    });

    it('El boton debe contener la palabra inicio', ()=>{
        expect(btnInicio).toHaveTextContent(/inicio/i)
    });

    it('El botÃ³n debe redirigir a la ruta de Home', ()=>{
        expect(btnInicio.getAttribute('href')).toBe('/home')
    })

})