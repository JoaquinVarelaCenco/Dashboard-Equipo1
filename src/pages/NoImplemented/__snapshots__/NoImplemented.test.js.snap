// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de la vista NoImplemented La vista debe renderizarse correctamente 1`] = `
<div>
  <div
    class="noImplemented"
  >
    <h3>
      <span>
        En
      </span>
       Construcción
    </h3>
    <p>
      Esta sección estara lista pronto
    </p>
    <div
      class="progress"
    >
      <div
        class="progress-bar"
      />
    </div>
    <a
      href="/home"
    >
      <button>
        Inicio
      </button>
    </a>
  </div>
</div>
`;
